packages/server/lib/browsers/electron.js

```diff
@@ -34,15 +34,36 @@ const tryToCall = function (win, method) {
   }
 }
 
-const getAutomation = function (win) {
+const _getAutomation = function (win, options) {
   const sendCommand = Bluebird.method((...args) => {
     return tryToCall(win, () => {
       return win.webContents.debugger.sendCommand
       .apply(win.webContents.debugger, args)
     })
   })
 
-  return CdpAutomation(sendCommand)
+  const automation = CdpAutomation(sendCommand)
+
+  if (!options.onScreencastFrame) {
+    // after upgrading to Electron 8, CDP screenshots can hang if a screencast is not also running
+    // workaround: start and stop screencasts between screenshots
+    // @see https://github.com/cypress-io/cypress/pull/6555#issuecomment-596747134
```

Comment:
will this be caught in an e2e test if removed?

Reply:
yep, this will fail: https://github.com/cypress-io/cypress/blob/19313dd24fb81dd533d42c2a2bc0ec94ea25ada8/packages/server/test/e2e/7_record_spec.coffee#L582-L593